contrib/google.golang.org/grpc/stats_server_test.go

```diff
@@ -78,3 +85,20 @@ func newServerStatsHandlerTestServer(statsHandler stats.Handler) (*rig, error) {
 		client:        NewFixtureClient(conn),
 	}, nil
 }
+
+func waitOnSpans(fetch func() []mocktracer.Span, timeout time.Duration) ([]mocktracer.Span, error) {
+	ticker := time.NewTicker(100 * time.Millisecond)
+	defer ticker.Stop()
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+	for {
+		select {
+		case <-ctx.Done():
+			return nil, errors.New("couldn't get spans in time")
+		case <-ticker.C:
+			if spans := fetch(); len(spans) > 0 {
+				return spans, nil
+			}
+		}
+	}
```

Comment:
NIT: This works, but using a ticker + context feels a bit overkill. What about a a simple for loop like this? Just takes 6 lines of code, so can be inlined into the test case itself.

```diff
diff --git a/contrib/google.golang.org/grpc/stats_server_test.go b/contrib/google.golang.org/grpc/stats_server_test.go
index 352c21df..68c84d61 100644
--- a/contrib/google.golang.org/grpc/stats_server_test.go
+++ b/contrib/google.golang.org/grpc/stats_server_test.go
@@ -38,12 +38,20 @@ func TestServerStatsHandler(t *testing.T) {
 	_, err = server.client.Ping(context.Background(), &FixtureRequest{Name: "name"})
 	assert.NoError(err)
 
-	spans := mt.FinishedSpans()
-	if len(spans) == 0 {
-		// we fetched finished spans too soon
-		spans, err = waitOnSpans(mt.FinishedSpans, 1*time.Second)
-		assert.NoError(err)
+	// Workaround for logical race: mt.FinishedSpans() might not return the span
+	// created by the ping right away. Keep checking for 1 second to avoid flaky
+	// test behavior.
+	//
+	// TODO: Ideally we could get an event from the fixtureServer when the span
+	// was created.
+	var spans []mocktracer.Span
+	for start := time.Now(); time.Since(start) < time.Second; {
+		if spans = mt.FinishedSpans(); len(spans) > 0 {
+			break
+		}
+		time.Sleep(10 * time.Millisecond)
 	}
+
 	assert.Len(spans, 1)
 
 	span := spans[0]
```

Reply:
Thanks! Reused the existing `waitForSpans` helper.